python/lsst/source/injection/inject_base.py

```diff
@@ -587,6 +592,11 @@ def _check_sources(self, injection_catalog, binary_flags):
         """
         self.config = cast(BaseInjectConfig, self.config)
 
+        # Exit early if there are no sources to inject.
+        if len(injection_catalog) == 0:
+            self.log.info("Catalog checking not applied to empty injection catalog.")
+            return injection_catalog
```

Comment:
I'm not sure if the `if` branch here will ever be true here given the check and return earlier, but shouldn't harm at least.

Reply:
Ah, I misread the positions these lines were placed at.